Plugin/Sales/Model/Spi/CreditmemoResource.php

```diff
@@ -193,12 +165,14 @@ public function aroundLoad(
         /** @var \Magento\Framework\Model\ResourceModel\Db\AbstractDb $resultEntity */
         $resultEntity = $proceed($entity, $value, $field);
 
-        // Load AvaTax extension attributes
+        // Load AvaTax record into extension attributes
         if ($this->avaTaxConfig->isModuleEnabled($entity->getStoreId())) {
 
-            // Get the AvaTax Attributes from the AbstractModel
-            $avataxIsUnbalanced = $entity->getData('avatax_is_unbalanced');
-            $baseAvataxTaxAmount = $entity->getData('base_avatax_tax_amount');
+            // Get the AvaTax record
+            /** @var CreditMemo $avataxRecord */
+            $avataxRecord = $this->avataxCreditMemo->loadByParentId($entity->getId());
+            $avataxIsUnbalanced = $avataxRecord->getData('is_unbalanced');
+            $baseAvataxTaxAmount = $avataxRecord->getData('base_avatax_tax_amount');
```

Comment:
If the magic getters and setters are defined on the class you shouldn't need to make calls to `getDate()` with the direct field name. You could just call `$avataxRecord->getBaseAvataxTaxAmount()`